src/backend/core/tests/test_api_utils_flat_to_nested.py

```diff
@@ -0,0 +1,109 @@
+"""Unit tests for the flat_to_nested utility function."""
+
+from core.api.utils import flat_to_nested
+
+
+def test_api_utils_flat_to_nested_empty_list():
+    """Test that an empty list returns an empty nested structure."""
+    # pylint: disable=use-implicit-booleaness-not-comparison
+    assert flat_to_nested([]) == []
+
+
+def test_api_utils_flat_to_nested_single_document():
+    """Test that a single document is returned as the only root element."""
+    documents = [{"id": "1", "depth": 0}]
+    expected = [{"id": "1", "depth": 0, "children": []}]
+    assert flat_to_nested(documents) == expected
+
+
+def test_api_utils_flat_to_nested_multiple_root_documents():
+    """Test that multiple root-level documents are correctly added to the root."""
```

Comment:
You can't have two roots element. IMO an exception should be raised if it's the case.

Reply:
yep I modified the method as discussed irl